vignettes/example_analysis_coxreg.Rmd

```diff
@@ -0,0 +1,380 @@
+      control = control,
+      data = data
+    )
+    rbind(covar_test, y)
+  }
+}
+```
+
+## Creating a Helper Function: `cached_model`
+
+Next, we will create a helper function, `cached_model`, which will be used within our analysis function to cache and return the fitted Cox regression model for the current covariate. The `df` argument will be directly inherited from the `df` argument passed to the analysis function, which contains the full dataset being analyzed. The `cov` argument will be the covariate that is being analyzed depending on the current row context. If the treatment effect is currently being analyzed, this value will be an empty string. The `cache_env` parameter will be an environment object which is used to store the model for the current covariate, also inherited directly from the analysis function. Of course, this function can also be run outside of the analysis function and will still cache and return a Cox regression model.
+
+Using these arguments, the `cached_model` function first checks if a model for the given covariate `cov` is already stored in the caching environment `cache_env`. If so, then this model is retrieved and returned by `cached_model`. If not, the model must be constructed. This is done by first constructing the model formula, `model_form`, starting with only the treatment effect (`ARM`) and adding a covariate effect if one is currently being analyzed. Then a Cox regression model is fit using `df` and the model formula, and this model is both returned and stored in the caching environment object as `cache_env[[cov]]`.
+
+```{r}
+cached_model <- function(df, cov, cache_env) {
+  ## Check if a model already exists for
+  ## `cov` in the caching environment
+  if (!is.null(cache_env[[cov]])) {
+    ## If model already exists, retrieve it from cache_env
+    model <- cache_env[[cov]]
+  } else {
+    ## Build model formula
+    model_form <- paste0("survival::Surv(AVAL, EVENT) ~ ARM")
+    if (length(cov) > 0) {
+      model_form <- paste(c(model_form, cov), collapse = " * ")
+    } else {
+      cov <- "ARM"
+    }
+    ## Calculate Cox regression model
+    model <- survival::coxph(
+      formula = stats::as.formula(model_form),
+      data = df,
+      ties = "exact"
+    )
+    ## Store model in the caching environment
+    cache_env[[cov]] <- model
+  }
+  model
+}
+```
+
+## Creating the Analysis Function: `a_cox_summary`
+
+With our data prepared and helper function created, we can proceed to construct our analysis function `a_cox_summary`, which will be used to populate all of the rows in our table. In order to be used to generate both data rows (for interaction effects) and content rows (for main effects), we must create a function that can be used as both `afun` in `analyze` and `cfun` in `summarize_row_groups`. Therefore, our function must accept either `x` or `df` as its first parameter and also accept the `labelstr` parameter.
```

Comment:
probably trim out the first part of this last sentence (all afuns and cfuns have to accept either x or df), the relevant bit is just the second half, that we make it accept labelstr so that it can be used in both contexts.